### PR TITLE
Add and use local CHIP project config headers

### PR DESCRIPTION
**Pull request by agners**
Instead of relying on the in-tree config headers in config/standalone/ pass a path to the project config to let the framework use our own config header files. This avoids using patches to adjust build specific configs.

Note that previously the builds in this repository used config/standalone/ instead of config/python/. The CHIP project config headers in config/python/ is what the script ./scripts/build_python.sh uses. This script is sometimes used to build the Python library during development. And was the reason why the wrong file got patched in #79.

There are minor differences between the standalone and python specific configs, specifically in CHIPProjectConfig.h. This PR adds a header file based on the config in config/python/, but with small adjustments.

Compared to the previous configuration, the effective change at build time should work out to:
- Dropping device specific configs (CHIP_DEVICE_CONFIG_*). These should not have an effect for our controller builds.
- Increae of CHIP_CONFIG_CONTROLLER_MAX_ACTIVE_DEVICES to 1024 (this time the new configruation gets applied correclty).
- Increase of CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS from 150 (standalone config) to 256 which feels more natural.

### CODE REVIEW

**Comment by agners** [comment]
FWIW, the size of a single operational session is 560 bytes. Since the Matter SDK uses static memory allocation only, this effectively increases memory usage by about ~500kb.

File: CHIPProjectConfig.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+#define CHIP_CONFIG_ENABLE_ARG_PARSER 1
+
+// Use a default pairing code if one hasn't been provisioned in flash.
+#define CHIP_DEVICE_CONFIG_USE_TEST_SETUP_PIN_CODE 20202021
+#define CHIP_DEVICE_CONFIG_USE_TEST_SETUP_DISCRIMINATOR 0xF00
+
+// Enable reading DRBG seed data from /dev/(u)random.
+// This is needed for test applications and the CHIP device manager to function
+// properly when CHIP_CONFIG_RNG_IMPLEMENTATION_CHIPDRBG is enabled.
+#define CHIP_CONFIG_DEV_RANDOM_DRBG_SEED 1
```
Comment (marcelveldt) [inline CHIPProjectConfig.h:41]: as these are specific to devices, cant we omit them ?

File: CHIPProjectConfig.h [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+//    WARNING: These options make it possible to circumvent basic Chip security functionality,
+//    including message encryption. Because of this they MUST NEVER BE ENABLED IN PRODUCTION BUILDS.
+//
+#define CHIP_CONFIG_SECURITY_TEST_MODE 0
+
+#define CHIP_CONFIG_ENABLE_UPDATE 1
+
+#define CHIP_SYSTEM_CONFIG_PACKETBUFFER_POOL_SIZE 0
+
+#define CHIP_CONFIG_DATA_MANAGEMENT_CLIENT_EXPERIMENTAL 1
```
Comment (marcelveldt) [inline CHIPProjectConfig.h:55]: what does this do ? It doenst sound that great that an experimental feature gets enabled :-)

**Review by marcelveldt — APPROVED** [review]

File: CHIPProjectConfig.h (hunk 1 above)
Comment (agners) [reply CHIPProjectConfig.h:41]: Most likely yes, but they used to be set previously via the standalone config see https://github.com/project-chip/connectedhomeip/blob/9d56786ed093473fdfb66d770a2e2338ab579d3e/config/standalone/CHIPProjectConfig.h#L40-L47.

It is a bit more convoluted, as the standalone config uses the catch all `CHIP_DEVICE_CONFIG_ENABLE_TEST_SETUP_PARAMS` config. But this essentially sets the same config as defined here (and is by default also enabled). So this should be a "same as before" config. 

FWIW, I intend to update this upstream and then backport later.

File: CHIPProjectConfig.h (hunk 2 above)
Comment (agners) [reply CHIPProjectConfig.h:55]: Similar here, it used to be there in the standalone config used previously. So staying with what has been used there.

**Pull request merged**